plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb.test.ts

```diff
@@ -244,6 +249,93 @@ describe('MongoDBInstrumentation', () => {
     });
   });
 
+  describe('when specifying a responseHook configuration', () => {
+    const dataAttributeName = 'mongodb_data';
+    beforeEach(() => {
+      memoryExporter.reset();
+    });
+
+    describe('with a valid function', () => {
+      beforeEach(() => {
+        create({
+          responseHook: (span: Span, result: CommandResult) => {
+            span.setAttribute(dataAttributeName, JSON.stringify(result.result));
+          },
+        });
+      });
+
+      it('should attach response hook data to the resulting span for insert function', done => {
+        const insertData = [{ a: 1 }, { a: 2 }, { a: 3 }];
+        const span = provider.getTracer('default').startSpan('insertRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.insertMany(insertData, (err, result) => {
+            span.end();
+            assert.ifError(err);
+            const spans = memoryExporter.getFinishedSpans();
+            const insertSpan = spans[0];
+
+            assert.strictEqual(
+              insertSpan.attributes[dataAttributeName],
+              JSON.stringify(result.result, Object.keys(result.result).sort())
+            );
+
+            memoryExporter.reset();
+            done();
+          });
+        });
+      });
+
+      it('should attach response hook data to the resulting span for find function', done => {
+        const span = provider.getTracer('default').startSpan('findRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.find({ a: 1 }).toArray((err, results) => {
+            span.end();
+            assert.ifError(err);
+            const spans = memoryExporter.getFinishedSpans();
+            const findSpan = spans[0];
+            const spanResult = JSON.parse(
+              findSpan.attributes[dataAttributeName]?.toString() || '{}'
```

Comment:
Current implementation gives the impression that `findSpan.attributes[dataAttributeName]` can be undefined (by using the `?` and setting a default value).
But that's something that the test should assert instead.
If the attribute is missing, we will just fail few lines below when trying to access `spanResult.cursor`.

Reply:
Added null validations inside the assertion - now if the `findSpan.attributes[dataAttributeName]` is undefined, the assertion will fail.

Reply:
cool, but still this code is written in a way that implies that `findSpan.attributes[dataAttributeName]` can be legitimately falsy, where the purpose of the test is exactly to assert that it is not.
I find it a bit confusing, but that's just a matter of style I suppose. If you prefer to have it this way, I'm good with it :)

Reply:
Just as the test above... if its empty - the test will fail.